### ISSUE TITLE
create an example for using container images as a reference

### ISSUE THREAD

**Issue by Alappin** [aws/aws-lambda-go#345]
**Is your feature request related to a problem? Please describe.**

Following the instructions in the docs e.g.

https://docs.aws.amazon.com/lambda/latest/dg/go-image.html 
https://hub.docker.com/r/amazon/aws-lambda-go

results in a docker image that does NOT run locally and when deployed. I don't know whether this container image feature is unstable or I am making trivial mistakes. It will be better if AWS provided a working github repository that demonstrates a complete solution.

**Describe the solution you'd like**

 I am requesting AWS to provide a working github repository that demonstrates a complete solution of using docker images with golang.

**Describe alternatives you've considered**

I have considered just using java but also had errors I could not resolve.

**Additional context**

A sample repository with a full solution similar to the one provided for dotent here would be appreciated.

https://github.com/aws/aws-lambda-dotnet/tree/master/LambdaRuntimeDockerfiles/dotnet5/sample

**Comment by bmoffatt** [comment]
Can you share the errors you ran into?

**Comment by Alappin** [comment]
Hi @bmoffatt 

I quickly created a sample github repo illustrating the issues. Please check this repository that contains a complete example based on the sample codes.

https://github.com/Alappin/alg

**Comment by mhart** [comment]
I showed an example of how to do this here:

https://hichaelmart.medium.com/using-container-images-with-aws-lambda-7ffbd23697f1

You need the runtime interface emulator to run/test it locally

**Comment by bmoffatt** [comment]
Hey @Alappin, thanks for the reproducer!

Your issue isn't with the docker build, it's with your test payload. I was able get a `200` response by changing it.

```
$ curl -XPOST "http://localhost:9000/2015-03-31/functions/function/invocations" -d '{"path": "/ping", "httpMethod": "GET"}' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   176  100   138  100    38  11500   3166 --:--:-- --:--:-- --:--:-- 14666
{
  "statusCode": 200,
  "headers": null,
  "multiValueHeaders": {
    "Content-Type": [
      "application/json; charset=utf-8"
    ]
  },
  "body": "{\"message\":\"pong\"}"
}
```

Your code shows that you're trying to test a function that integrates with API gateway. When API Gateway receives an HTTP request in the cloud, it translates this into a JSON object, and then sends it to Lambda. Ditto with the response. You can find an example of what this JSON can look like in this project's tests. The `404` is happening because of the logic in the `github.com/awslabs/aws-lambda-go-api-proxy` library, which re-hydrates this JSON into an `http.Request`. Because your test JSON defines no path or method, the gin router understandably returns a `404`

To test API Gateway + Lambda locally, you can either produce the correct test JSON, or use a tool like the AWS SAM CLI, which provides a local API Gateway emulator https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-start-api.html

**Comment by Alappin** [comment]
Hi @bmoffatt 

Indeed, figuring out the way to do the request was one of my main problems. Thanks for looking into it and clarifying the request approach.

I also figured out why it was not running when deployed, that was because I had the API gateway integration set as V2 in the console, so I had to change that to V1 for it to actually work. The aws lambda go events have

`events.APIGatewayProxyRequest`

I could not find the equivalent 

`events.APIGateway{V2}ProxyRequest`

to use instead which I just changed it on the api gateway settings.
